FirebaseDatabase/Tests/Unit/Swift/DatabaseAPITests.swift

```diff
@@ -0,0 +1,602 @@
+        // ...
+      }
+    databaseReference
+      .observeSingleEvent(of: DataEventType
+        .childMoved) { (dataSnapshot: DataSnapshot, optionalString: String?) in
+        // ...
+      }
+    #if swift(>=5.5)
+      if #available(iOS 15.0, macOS 12.0, macCatalyst 15.0, tvOS 15.0, *) {
+        // async/await is a Swift 5.5+ feature available on iOS 15+
+        async {
+          do {
+            _ = try await databaseReference
+              .observeSingleEventAndPreviousSiblingKey(of: DataEventType.value) as (DataSnapshot,
+                                                                                    String?)
+          } catch {
+            // ...
+          }
+        }
+      }
+    #endif // swift(>=5.5)
+    databaseReference
+      .observeSingleEvent(of: DataEventType.childChanged) { (dataSnapshot: DataSnapshot) in
+        // ...
+      } withCancel: { (error: Error) in
+        // ...
+      }
+    databaseReference
+      .observeSingleEvent(of: DataEventType
+        .childMoved) { (dataSnapshot: DataSnapshot, optionalString: String?) in
+        // ...
+      } withCancel: { (error: Error) in
+        // ...
+      }
+    // Get data
+    databaseReference.getData { (optionalError: Error?, dataSnapshot: DataSnapshot) in
+      // ...
+    }
+    #if swift(>=5.5)
+      if #available(iOS 15.0, macOS 12.0, macCatalyst 15.0, tvOS 15.0, *) {
+        // async/await is a Swift 5.5+ feature available on iOS 15+
+        async {
+          do {
+            _ = try await databaseReference.getData() as DataSnapshot
+          } catch {
+            // ...
+          }
+        }
+      }
+    #endif // swift(>=5.5)
+    // Remove Observers
+    databaseReference.removeObserver(withHandle: 0 as DatabaseHandle)
+    databaseReference.removeAllObservers()
+    // Keep Synced
+    databaseReference.keepSynced(true as Bool)
+    // Limited Views of Data
+    _ = databaseReference.queryLimited(toFirst: 0 as DatabaseHandle) as DatabaseQuery
+    _ = databaseReference.queryLimited(toLast: 1 as DatabaseHandle) as DatabaseQuery
+    _ = databaseReference.queryOrdered(byChild: "key" as String) as DatabaseQuery
+    _ = databaseReference.queryOrderedByKey() as DatabaseQuery
+    _ = databaseReference.queryOrderedByPriority() as DatabaseQuery
+    _ = databaseReference.queryStarting(atValue: "value" as Any?) as DatabaseQuery
+    _ = databaseReference.queryStarting(
+      atValue: "value" as Any?,
+      childKey: "key" as String?
+    ) as DatabaseQuery
+    _ = databaseReference.queryStarting(afterValue: "value" as Any?) as DatabaseQuery
+    _ = databaseReference.queryStarting(
+      afterValue: "value" as Any?,
+      childKey: "key" as String?
+    ) as DatabaseQuery
+    _ = databaseReference.queryEnding(atValue: "value" as Any?) as DatabaseQuery
+    _ = databaseReference.queryEnding(
+      atValue: "value" as Any?,
+      childKey: "key" as String?
+    ) as DatabaseQuery
```

Comment:
`DatabaseReference` inherits from `DatabaseQuery` the following methods:
- `open func queryEnding(beforeValue endValue: Any?) -> DatabaseQuery`
- `open func queryEnding(beforeValue endValue: Any?, childKey: String?) -> DatabaseQuery`

These (and possibly other similar methods) are not currently tested since they do not show up in the generated Swift 5 Interface for FIRDatabaseReference.h, but should they?

Reply:
Probably not necessary - at least in the first version